src/pages/PromptPage/PromptPage.jsx

```diff
@@ -1,108 +1,45 @@
-  const [isEditing, setIsEditing] = useState(false);
-
-  const getOpenAIResponse = async () => {
-    setPreviewText("Loading...");
-    requestObj.messages[1].content = inputText;
-
-    try {
-      const responseContent = await getResponseContent(requestObj);
-
-      // need to parse JSON to return an object we can work with
-      const parsedContent = JSON.parse(responseContent.content);
-
-      // // If responseContent has no content, setAiResponseContent to error message
-      setPreviewText(
-        parsedContent.content
-          ? parsedContent.content
-          : `Error: ${responseContent}`
-      );
-    } catch (error) {
-      setPreviewText("An error occured. Please try again.");
-    }
-  };
+  const [activeTab, setActiveTab] = useState("edit");
 
   const handleInputChange = (e) => {
     setInputText(e.target.value);
   };
 
-  const handleGenerateButtonClick = () => {
-    if (inputText) {
-      getOpenAIResponse();
-    } else {
-      alert("Please add input!");
-    }
-  };
-
-  const handleToggleEditing = () => {
-    setIsEditing(!isEditing);
-  };
-
-  // placeholder to handle LinkedIn posting
-  const handlePostButtonClick = () => {};
-
-  // conditional rendering refactored out of main return statement
-  const previewContent = () => {
-    return isEditing ? (
-      <InputBox
-        value={previewText}
-        onChange={(e) => setPreviewText(e.target.value)}
-        className="promptpage__preview-box"
-      />
-    ) : (
-      <PreviewBox
-        previewText={previewText}
-        className="promptpage__preview-box"
-        onClickEdit={handleToggleEditing}
-      />
+  const renderActiveTab = () => {
+    return (
+      <div className="promptpage">
+        {activeTab === "edit" && (
+          <EditPost
+            inputText={inputText}
+            handleInputChange={handleInputChange}
+            setActivePage={setActiveTab}
+            setPreviewText={setPreviewText}
+          />
+        )}
+        {activeTab === "review" && (
+          <ReviewPost
+            previewText={previewText}
+            setPreviewText={setPreviewText}
+            setActivePage={setActiveTab}
+          />
+        )}
+      </div>
     );
   };
 
   return (
-    <div className="promptpage">
-      <div className="promptpage__input-container">
-        <InputBox
-          value={inputText}
-          onChange={handleInputChange}
-          placeholder="Help me write a professional sounding post about my upcoming community event."
-          className="promptpage__input-box"
-        />
-        <Button className="generate" onClick={handleGenerateButtonClick}>
-          Generate
-        </Button>
-      </div>
-      <div className="promptpage__preview-container">
-        {previewContent()}
-        <Button className="post" onClick={handlePostButtonClick}>
-          Post
-        </Button>
-      </div>
-    </div>
+    <>
+      <PromptPageTabs activeTab={activeTab} setActiveTab={setActiveTab} />
```

Comment:
Do we need to pass setActiveTab here? Is it being used in PromptPageTabs?

Reply:
You're right we don't need it. I'll remove!